docs/adrs/0549-composite-run-steps.md

```diff
@@ -0,0 +1,321 @@
+
+In the child action (in this example, this is the `action.yml`), it starts with a clean slate (in other words, no imposing if conditions). Similar to the logic in the paragraph above, `echo "I will run, as my current scope is succeeding"` will run since the `if` condition checks if the previous steps **within this composite action** has not failed. `run: echo "I will not run, as my current scope is now failing"` will not run since the previous step resulted in an error and by default, the if expression is set to `success()` if the if condition is not set for a step.
+
+
+What if a step has `cancelled()`? We do the opposite of our approach above if `cancelled()` is used for any of our composite run steps. We will cancel any step that has this condition if the workflow is cancelled at all.
+
+#### Exposing Parent's If Condition to Children Via a Variable
+It would be nice to have a way to access information from a parent's if condition. We could have a parent variable that is contained in the context similar to other context variables `github`, `strategy`, etc.:
+
+Example `workflow.yml`
+
+```yaml
+steps:
+  - run: exit 1
+  - uses: user/composite@v1
+    if: always()
+```
+
+Example `user/composite/action.yml`
+
+```yaml
+runs:
+  using: "composite"
+  steps:
+    - run: echo "preparing the slack bot..."  # <--- This will run, as nothing has failed within the composite yet
+    - run: slack.post("All builds passing, ready for a deploy")  # <-- this will not run, as the parent fails
+      if: ${{ parent.success() }}
+    - run: slack.post("A failure has happened, fix things now", alert=true)  # <--- This will run, as the parent fails
+      if: ${{ parent.failure() }}
+```
+    
+## Timeout-minutes
+
+Example `workflow.yml`:
+
+```yaml
+steps: 
+  - id: bar
+    uses: user/test@v1
+    timeout-minutes: 50
+```
+
+Example `user/composite/action.yml`:
+
+```yaml
+runs:
+  using: "composite"
+  steps: 
+    - id: foo1
+      run: echo test 1
+      timeout-minutes: 10
+    - id: foo2
+      run: echo test 2
+    - id: foo3
+      run: echo test 3
+      timeout-minutes: 10
+```
+
+**TODO: This timeout-minutes condition implementation is up to discussion.** 
+
+A composite action in its entirety is a job. You can set both timeout-minutes for the whole composite action or its steps as long as the the sum of the `timeout-minutes` for each composite action step that has the attribute `timeout-minutes` is less than or equals to `timeout-minutes` for the composite action. There is no default timeout-minutes for each composite action step. 
+
+If the time taken for any of the steps in combination or individually exceed the whole composite action `timeout-minutes` attribute, the whole job will fail. If an individual step exceeds its own `timeout-minutes` attribute but the total time that has been used including this step is below the overall composite action `timeout-minutes`, the individual step will fail but the rest of the steps will run. 
```

Comment:
` the individual step will fail but the rest of the steps will run.` -> the rest of the steps will run base on condition.